### PR TITLE
M3-6035: Add type-to-confirm to destructive actions

### PR DESCRIPTION
**Pull request by hana-linode**
## Description 📝
Add new TypeToConfirmDialog component and add type-to-confirm to the following dialogs:

1. Delete Linode on the landing/settings page
2. Delete Attached Volume
3. Delete Unattached Volume
4. Delete NodeBalancer on the landing/settings page

## Preview 📷
image

## How to test 🧪
Confirm that the dialogs mentioned in the description have type-to-confirm

**How do I run relevant unit or e2e tests?**
```
yarn test TypeToConfirmDialog
```

### CODE REVIEW

File: packages/manager/src/features/NodeBalancers/NodeBalancersLanding/NodeBalancersLanding.tsx [hunk 1]
```diff
@@ -110,6 +117,7 @@ export class NodeBalancersLanding extends React.Component<
     submitting: false,
     open: false,
     errors: undefined,
+    confirmText: '',
```
[the remark this answers is not in the document]
Comment (hana-linode) [reply packages/manager/src/features/NodeBalancers/NodeBalancersLanding/NodeBalancersLanding.tsx]: Good catch! This seems to be an existing bug but I fixed the occurrences you mentioned. There's probably more places this is occurring in. I'll create a new ticket to refactor the other components to use the new TypeToConfirmDialog component which should fix the issue

File: packages/manager/src/features/Volumes/DestructiveVolumeDialog.tsx [hunk 2]
```diff
@@ -119,66 +109,40 @@ export const DestructiveVolumeDialog = (props: Props) => {
...
-          power off the Linode first or make sure it isn't currently writing to
-          the volume before continuing. If this volume is currently mounted,
-          detaching it could cause your Linode to restart.
+          power off the Linode first or make sure it isn&rsquo;t currently
+          writing to the volume before continuing. If this volume is currently
+          mounted, detaching it could cause your Linode to restart.
         </Typography>
       )}
       <Typography>
         Are you sure you want to {mode} this Volume
```
Comment (mjac0bs) [inline packages/manager/src/features/Volumes/DestructiveVolumeDialog.tsx:143]: It looks like the text here is consistent for both volume detach and delete actions on the Volumes landing and Linode details page, but it is the older text ("are you sure you want...") we replaced for Linode delete. Thoughts on also making that change here?

**Review by mjac0bs — APPROVED** [review]
Nice, sounds like a good plan with the additional follow up task for any other TTC instances. 👍  Confirming that these updates fix the issue with persisting `confirmText` in the places I mentioned above and the unit test looks good. Leaving one non-blocking comment about warning text on the volume confirm dialogs.

**Review by bnussman-akamai — APPROVED** [review]
Nice work on this!

**Pull request merged**